### DISCUSSION TITLE
Getting key signature, time signature and different instruments from a midifile

### DISCUSSION THREAD

**Discussion by GardSveen** [melanchall/drywetmidi#159] [Q&A] [answered]
Hi. I am wondering how i can get the key signature and time signature from a midifile. And i was also wondering how to get the different instruments and the notes belonging to them(for example getting the notes in the treble and bass staff from piano notation into 2 different arrays).

**Comment by melanchall** [comment, marked as answer]
Hi,

First of all, there are no such things as _treble and bass staff_ in a MIDI file. MIDI file contains track chunks. We can get of course Key Signature and Time Signature events (there can be multiple such events in a MIDI file) and then you can decide by yourself where treble staff and bass one.

Please give me a MIDI file you want to retrieve the info from and I'll be able to say more. And also please say where bass and treble staffs in the file.

**Reply by GardSveen** [reply]
Hi i figured it out after reading more about track chunks. Thanks for the help.

**Comment by GardSveen** [comment]
Sorry for a bad example. What i am having trouble with is taking a midi file made in for example Sibelius with multiple instruments(for example Piano and violin). And then getting the notes belonging to them. I have made a function that uses GetNotes() and returns an array with all the notes(see code), but i want to have different arrays for different instruments. Is this posible?

public Melanchall.DryWetMidi.Interaction.Note[] GetMidiNotesArray(){
        var notes = midiFile.GetNotes();

        var array = new Melanchall.DryWetMidi.Interaction.Note[notes.Count];
        notes.CopyTo(array, 0);

        return array;
}

[Example Defferent instruments.mid.zip](https://github.com/melanchall/drywetmidi/files/7935876/Example.Defferent.instruments.mid.zip)
